rules/TypeDeclaration/TypeInferer/ReturnTypeInferer.php

```diff
@@ -157,15 +157,15 @@ static function (Node $subNode): bool {
     private function resolveBenevolentUnionTypeInteger(
         ClassMethod|Function_|Closure|ArrowFunction $functionLike,
         UnionType $unionType
-    ): UnionType|IntegerType {
+    ): Type {
```

Comment:
I think this is OK as it's a private method with only one usage in another method, usage that is wrapped by `instanceof UnionType`, and that other method early return the result of this one only if `isInteger()`

Reply:
see https://github.com/rectorphp/rector-src/blob/7515f5289810f32d25a8ea2ac934d1c3ff234351/rules/TypeDeclaration/TypeInferer/ReturnTypeInferer.php#L147-L152